lib/cesiumGeometryToGltfPrimitive.js

```diff
@@ -1,10 +1,89 @@
 'use strict';
+var findAccessorMinMax = require('./findAccessorMinMax');
 var writeAccessor = require('./writeAccessor');
+var mergeBuffers = require('./mergeBuffers');
+var uninterleaveAndPackBuffers = require('./uninterleaveAndPackBuffers');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var WebGLConstants = Cesium.WebGLConstants;
+
 
 module.exports = cesiumGeometryToGltfPrimitive;
 
+function createAttributeSemantic(gltf, primitive, semantic, attributeLength) {
+    var type;
+    var bytesPerComponent;
+
+    switch (semantic.replace(/([0-9]|_)/g, '')) {
+        case 'POSITION':
+            type = 'VEC3';
+            bytesPerComponent = 8;
+            break;
+        case 'NORMAL':
+            type = 'VEC3';
+            bytesPerComponent = 4;
+            break;
+        case 'TEXCOORD':
+            type = 'VEC2';
+            bytesPerComponent = 4;
+    }
+
+    var componentsPerType = numberOfComponentsForType(type);
+
+    var bufferLength = attributeLength * componentsPerType * bytesPerComponent;
+    var buffer = new Buffer(bufferLength);
+
+    var lowercaseSemantic = semantic.toLowerCase();
+    var bufferId = 'buffer_' + lowercaseSemantic + '_generated';
+    var bufferViewId = 'bufferView_' + lowercaseSemantic + '_generated';
+    var accessorId = 'accessor_' + lowercaseSemantic + '_generated';
+
+    gltf.buffers[bufferId] = {
+        byteLength : bufferLength,
+        type : 'arraybuffer',
+        extras : {
+            _pipeline : {
+                source : buffer,
+                extension : '.bin'
+            }
+        }
+    };
+
+    gltf.bufferViews[bufferViewId] = {
+        buffer : bufferId,
+        byteLength : bufferLength,
+        byteOffset : 0,
+        target : WebGLConstants.ARRAY_BUFFER
+    };
+
+    var accessor = {
+        bufferView : bufferViewId,
+        byteOffset : 0,
+        byteStride : 0,
+        componentType : WebGLConstants.FLOAT,
+        count : attributeLength,
+        type : type
+    };
+
+    gltf.accessors[accessorId] = accessor;
+
+    // Find min/max for normal accessor
+    var minMax = findAccessorMinMax(gltf, accessor);
+    accessor.min = minMax.min;
+    accessor.max = minMax.max;
+
+    primitive.attributes[semantic] = accessorId;
+    mergeBuffers(gltf, 'buffer_0');
+    uninterleaveAndPackBuffers(gltf);
```

Comment:
The merge and interleave should happen at the end of `cesiumGeometryToGltfPrimitive` just so we don't call this for every attribute.